beacon_node/store/src/errors.rs

```diff
@@ -38,27 +39,43 @@ pub enum Error {
     /// State reconstruction failed because it didn't reach the upper limit slot.
     ///
     /// This should never happen (it's a logic error).
-    StateReconstructionDidNotComplete,
+    StateReconstructionLogicError,
     StateReconstructionRootMismatch {
         slot: Slot,
         expected: Hash256,
         computed: Hash256,
     },
+    MissingStateRoot(Slot),
+    MissingState(Hash256),
+    MissingGenesisState,
+    MissingSnapshot(Slot),
+    NoBaseStateFound(Hash256),
```

Comment:
Unused as well

Reply:
Middle two are used, but removed the first and last in https://github.com/sigp/lighthouse/pull/5978/commits/c4d6ef3544b529bbbc8a919916165e6e976eb1fb